File: June2021/python/reverse_linked_list_2.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def reverseBetween(self, head, left, right):
        
        node_val = []
        temp = head
        
        left -=1
        right -=1
        
        c = 0
        
        while c != right + 1:     
            if c >= left and c <=right:
                node_val.append(temp.val)
            c+=1
            temp = temp.next
            
        c2 = 0
        temp_2 = head
        
        while c2 != right+1:
            if c2 >= left and c2 <=right:
                temp_2.val = node_val.pop(-1)
            c2+=1
            temp_2 = temp_2.next
            
        return head

File: July2022/C++/reverseLinkedList2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        
        if(!head) return head;
        
        vector<int> v;
        ListNode* h1 = head;
        while(h1)
        {  
            v.push_back(h1->val);
            h1 = h1 -> next;
        }

        reverse(v.begin()+left-1, v.begin()+right);
        
        ListNode* ans = new ListNode(0);
        ListNode* cur = ans;

        for(int i=0;i<v.size();i++)
        {
            ListNode* temp = new ListNode(0);
            temp->val = v[i];
            cur -> next = temp;
            cur = cur ->next;
        }
        return ans->next;
        
    }
};
